### PR TITLE
[WIP]Bot webhooks

### PR DESCRIPTION
**Pull request by softbobo**
This is long overdue. Implemention MVP of webhooks for the bot. all functions are deactivated for now in an effort to refactor the project. Hope to progress in the next few days so we can bring new features soon. Early Feedback is always welcome.

### CODE REVIEW

File: docker-compose.yaml [hunk 1]
```diff
@@ -3,7 +3,18 @@ services:
 
   website:
     build: ./serv
-    image: server:latest
+    container_name: serv
     ports:
       - "80:80"
       - "443:443"
+<<<<<<< HEAD
```
Comment (obitech) [inline docker-compose.yaml:10]: This needs some formatting

File: docker-compose.yaml [hunk 2]
```diff
@@ -3,7 +3,18 @@ services:
 
   website:
     build: ./serv
-    image: server:latest
+    container_name: serv
```
Comment (obitech) [inline docker-compose.yaml:6]:
```suggestion
    container_name: server
```

File: serv/Dockerfile [hunk 3]
```diff
@@ -5,4 +5,8 @@ COPY nginx.conf /etc/nginx
 COPY fullchain.pem /var/www/certs/
 COPY privkey.pem /var/www/certs/
 
+<<<<<<< HEAD
```
Comment (obitech) [inline serv/Dockerfile:8]: Formatting

File: serv/nginx.conf [hunk 4]
```diff
@@ -4,8 +4,17 @@ events {
 
 http {
 
+<<<<<<< HEAD
```
Comment (obitech) [inline serv/nginx.conf:7]: Formatting

File: serv/nginx.conf [hunk 5]
```diff
@@ -4,8 +4,17 @@ events {
...
+    upstream bot {
+        server bot:8443;
+    }
+
+    server {
+	    listen 443 ssl;
+=======
     server {
 	listen 443 ssl;
+>>>>>>> origin/main
```
Comment (obitech) [inline serv/nginx.conf:17]: Formatting

File: serv/nginx.conf [hunk 6]
```diff
@@ -14,10 +23,20 @@ http {
         ssl_certificate_key /var/www/certs/privkey.pem;
         ssl_protocols       TLSv1 TLSv1.1 TLSv1.2;
 	
+<<<<<<< HEAD
```
Comment (obitech) [inline serv/nginx.conf:26]: Formatting

File: bot/wuestbot.py [hunk 7]
```diff
@@ -1,6 +1,374 @@
...
+# Handle '/start' and '/help'
+@bot.message_handler(commands=['help', 'start'])
+def send_welcome(message):
+    bot.reply_to(message, message.text)
+
+# Handle all other messages
+@bot.message_handler(func=lambda message: True, content_types=['text'])
+def echo_message(message):
+    bot.reply_to(message, message.text)
+#
```
Comment (obitech) [inline bot/wuestbot.py:50]: Let's delete this whole comment, we can always fetch the code from previous commits

File: bot/wuestbot.py [hunk 8]
```diff
@@ -1,6 +1,374 @@
...
+
+img_dir = os.path.abspath('../bot/app/img')
+
+# Process webhook calls
+@app.route(f'/bot', methods=['POST'])
+def webhook():
+    if flask.request.headers.get('content-type') == 'application/json':
+        json_string = flask.request.get_data().decode('utf-8')
+        update = telebot.types.Update.de_json(json_string)
+        bot.process_new_updates([update])
```
Comment (obitech) [inline bot/wuestbot.py:36]: What does `process_new_updates` do exacty? 🤔

File: bot/Dockerfile [hunk 9]
```diff
@@ -5,4 +5,6 @@ ADD . /app
 
 RUN pip install -r requirements.txt
 
-CMD ./start.sh
\ No newline at end of file
+EXPOSE 8443
+
+CMD exec gunicorn wuestbot:app -b 0.0.0.0:8443 -w 1 --access-logfile -
```
Comment (obitech) [inline bot/Dockerfile:10]: Since `start.sh` is not called, we won't be in the virtual environment right?

File: bot/wuestbot.py (hunk 8 above)
Comment (softbobo) [reply bot/wuestbot.py:36]: Takes the json objects from the webhook, sorts it into message categories and hands those over to specific handler functions. basically a big bunch of [if statements](https://github.com/eternnoir/pyTelegramBotAPI/blob/87bce0bce149bc6b3aa67f1252a3115273d94c3f/telebot/__init__.py#L325)

Comment (obitech) [reply bot/wuestbot.py:36]: Ah it's a library function

**Review by softbobo — COMMENTED** [review]
LGTM. Bot runs on Webhooks now and has a bit of new content before ROUTINES. more features to be added in nwe pr

**Pull request merged**